src/backends/torch/torchlib.cc

```diff
@@ -175,17 +175,44 @@ namespace dd
             tmpl, template_params, inputc));
 
     if (_native)
-      if (!tmodel._native.empty())
-        torch::load(_native, tmodel._native, device);
+      {
+        _logger->info("created net using template " + tmpl);
+        if (!tmodel._native.empty())
+          {
+            _logger->info("loading " + tmodel._native);
+            try
+              {
+                torch::load(_native, tmodel._native, device);
+              }
+            catch (std::exception &e)
+              {
+                _logger->warn("unable to load " + tmodel._native);
+                throw;
+              }
+          }
+      }
 
     if (_graph)
       {
         std::vector<long int> dims = inputc._dataset.datasize(0);
         dims.insert(dims.begin(), 1); // dummy batch size
         _graph->finalize(dims);
+        if (_graph->needs_reload())
+          _logger->info("net was reallocated due to input dim changes");
         // reload params after finalize
-        if (!tmodel._traced.empty())
-          torch::load(_graph, tmodel._traced, _device);
+        if (!tmodel._traced.empty() && _graph->needs_reload())
+          {
+            _logger->info("loading " + tmodel._traced);
```

Comment:
I believe we may want a function for this, like `graph_model_load()` to make code lighter.

Reply:
ok